lib/bridge/MatrixHandler.js

```diff
@@ -1486,6 +1517,89 @@ MatrixHandler.prototype._onUserQuery = Promise.coroutine(function*(req, userId)
     yield this.ircBridge.getMatrixUser(ircUser);
 });
 
+MatrixHandler.prototype._textForReplyEvent = Promise.coroutine(function*(event, ircRoom) {
+    const REPLY_REGEX = /> <(@.*:.*)>(.*)\n\n(.*)/;
+    const REPLY_NAME_MAX_LENGTH = 12;
+    const eventId = event.content["m.relates_to"]["m.in_reply_to"].event_id;
+    const match = REPLY_REGEX.exec(event.content.body);
+    if (match.length !== 4) {
+        return;
+    }
+
+    let rplName;
+    let rplSource;
+    const rplText = match[3];
+    if (!this._eventCache.has(eventId)) {
+        // Fallback to fetching from the homeserver.
+        try {
+            const eventContent = yield this.ircBridge.getAppServiceBridge().getIntent().getEvent(
+                event.room_id, eventId
+            );
+            rplName = eventContent.sender;
+            if (typeof(eventContent.content.body) !== "string") {
+                throw Error("'body' was not a string.");
+            }
+            const isReply = event.content["m.relates_to"] &&
+                event.content["m.relates_to"]["m.in_reply_to"];
+            if (isReply) {
+                const sourceMatch = REPLY_REGEX.exec(eventContent.content.body);
+                rplSource = sourceMatch.length === 4 ? sourceMatch[3] : event.content.body;
+            }
+            else {
+                rplSource = eventContent.content.body;
+            }
+            rplSource = rplSource.substr(0, REPLY_SOURCE_MAX_LENGTH);
+            this._eventCache.set(eventId, {sender: rplName, body: rplSource});
+        }
+        catch (err) {
+            // If we couldn't find the event, then frankly we can't
+            // trust it and we won't treat it as a reply.
+            return {
+                formatted: rplText,
+                reply: rplText,
+            };
+        }
+    }
+    else {
+        rplName = this._eventCache.get(eventId).sender;
+        rplSource = this._eventCache.get(eventId).body;
+    }
+
+    // Get the first non-blank line from the source.
+    const lines = rplSource.split('\n').filter((line) => !/^\s*$/.test(line))
+    if (lines.length > 0) {
+        // Cut to a maximum length.
+        rplSource = lines[0].substr(0, REPLY_SOURCE_MAX_LENGTH);
+        // Ellipsis if needed.
+        if (rplSource.length > REPLY_SOURCE_MAX_LENGTH) {
```

Comment:
This condition won't be activated because the instruction directly above trims it to a safe length.

Reply:
oh wow, that's pretty silly